domain/src/main/java/org/fao/geonet/domain/IMetadata.java

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (C) 2001-2016 Food and Agriculture Organization of the
+ * United Nations (FAO-UN), United Nations World Food Programme (WFP)
+ * and United Nations Environment Programme (UNEP)
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or (at
+ * your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ *
+ * Contact: Jeroen Ticheler - FAO - Viale delle Terme di Caracalla 2,
+ * Rome - Italy. email: geonetwork@osgeo.org
+ */
+package org.fao.geonet.domain;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.persistence.Basic;
+import javax.persistence.Column;
+import javax.persistence.Embedded;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Lob;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Transient;
+
+import org.apache.lucene.document.Document;
+import org.fao.geonet.utils.Xml;
+import org.hibernate.annotations.Type;
+import org.jdom.Element;
+import org.jdom.JDOMException;
+import org.jdom.output.Format;
+import org.jdom.output.XMLOutputter;
+
+import com.vividsolutions.jts.util.Assert;
+
+/**
+ * An entity representing a metadata object in the database. The xml, groups and operations are lazily loaded so accessing then will need to
+ * be done in a thread that has a bound EntityManager.
+ * 
+ * Also they can trigger database access if they have not been cached and therefore can cause slowdowns so they should only be accessed in
+ * need.
+ *
+ * All classes/tables implemented will share by default the same sequence for the ID. So you can have different kinds of metadata (like
+ * original and draft versioning) and the id will be unique on all the tables at the same time.
+ *
+ * @author María Arias de Reyna
+ */
+@MappedSuperclass
+public abstract class IMetadata extends GeonetEntity {
```

Comment:
The name `IMetadata` gives the impression that defines a Java Interface, maybe another name like `BaseMetadata` or `AbstractMetadata` can be better?

Reply:
Maybe AbstractRecord? Ok, will change it. I had a very Java vision here, yes.